src/contracts/atlas/GasAccounting.sol

```diff
@@ -237,71 +249,167 @@ abstract contract GasAccounting is SafetyLocks {
     /// @dev This function adjusts the claims, withdrawals, deposits, and surcharges based on the gas used by the
     /// transaction.
     /// @param ctx Context struct containing relavent context information for the Atlas auction.
-    /// @param winningSolver The address of the winning solver.
     function _settle(
         Context memory ctx,
-        address winningSolver
+        uint256 solverGasLimit
     )
         internal
-        returns (uint256 claimsPaidToBundler, uint256 netGasSurcharge)
+        returns (uint256 claimsPaidToBundler, uint256 netAtlasGasSurcharge)
     {
         // NOTE: If there is no winningSolver but the dApp config allows unfulfilled 'successes,' the bundler
         // is treated as the Solver.
 
+        // If a solver won, their address is still in the _solverLock
+        address _winningSolver = address(uint160(_solverLock));
+
         // Load what we can from storage so that it shows up in the gasleft() calc
         uint256 _surcharge = cumulativeSurcharge;
-        uint256 _claims = claims;
+
         uint256 _withdrawals = withdrawals;
         uint256 _deposits = deposits;
+        (uint256 _claims, uint256 _writeoffs) = _getAdjustedClaimsAndWriteoffs(ctx, solverGasLimit);
+
+        netAtlasGasSurcharge =
+            (_claims * ATLAS_SURCHARGE_RATE) / (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE);
+
+        // Update the stored cumulative Surcharge
+        cumulativeSurcharge = _surcharge + netAtlasGasSurcharge;
 
-        // Remove any unused gas from the bundler's claim.
-        // TODO: consider penalizing bundler for too much unused gas (to prevent high escrow requirements for solvers)
-        uint256 gasRemainder = gasleft() * tx.gasprice;
-        gasRemainder = gasRemainder * (SURCHARGE_SCALE + SURCHARGE_RATE) / SURCHARGE_SCALE;
-        _claims -= gasRemainder;
+        // Handle the settlement accounting logs
+        if (!ctx.solverSuccessful) {
+            // CASE: No solver was successful
 
-        if (_deposits < _claims + _withdrawals) {
-            // CASE: in deficit, subtract from bonded balance
-            uint256 amountOwed = _claims + _withdrawals - _deposits;
-            uint256 deficit = _assign(winningSolver, amountOwed, true, false);
+            // Ignore assigning to the winningSolver, just refund the bet deposits to the bundler if applicable
+            if (_deposits < _withdrawals + netAtlasGasSurcharge) {
+                // NOTE: We assume the bundler is fully on the hook for the gas, so we ignore claims and writeoffs
+                // and focus exclusively on deposits and withdrawals.
+                // The "+ netAtlasGasSurcharge" forces Bundler to pay the surcharge
+                revert InsufficientTotalBalance(_withdrawals + netAtlasGasSurcharge - _deposits);
+            }
+            claimsPaidToBundler = _deposits - _withdrawals;
+        } else if (_winningSolver == ctx.bundler) {
```

Comment:
but ctx.bundler refers to the "caller" not the actual atlas bundler (tx.origin)?

Reply:
correct.